extkeys/mnemonic.go

```diff
@@ -128,8 +133,8 @@ func (m *Mnemonic) MnemonicPhrase(strength, language Language) (string, error) {
 	// With more entropy security is improved but the sentence length increases.
 	// We refer to the initial entropy length as ENT. The recommended size of ENT is 128-256 bits.
 
-	if strength%32 > 0 && strength < 128 && strength > 256 {
-		return "", errors.New("The mnemonic must encode entropy in a multiple of 32 bits, The recommended size of ENT is 128-256 bits")
+	if strength%32 > 0 || strength < 128 || strength > 256 {
```

Comment:
I like it. Also, if we don't export `Strength`, but we just use `type Strength int`, it would be impossible to call the API with another value like `Strength(123)`. 

Does it make sense @mandrigin

Reply:
?

Reply:
absolutely, we shouldn't export what we don't need to.

Reply:
actually I thought go behaves in a different way.
Even if we have:

```
type strength int

const S1 = strength(128)

func foo(s strength) {}
```

You can still call `foo` with another number: `foo(111)`.

So in the end it wouldn't change so much, but maybe is it still more readable? @mandrigin

Reply:
Yeah, it doesn't reduce the need for a check, but it makes the API a bit harder to misuse.
To make is stronger, you can, of course make 
```go
type strength struct {
value int
}
```

but I think that's an overkill.

Reply:
Hey @pilu would be nice to have this conditional as a method itself, allowing a readable code:

```
	// The mnemonic must encode entropy in a multiple of 32 bits.
	// With more entropy security is improved but the sentence length increases.
	// We refer to the initial entropy length as ENT. The recommended size of ENT is 128-256 bits.
        func (m *Mnemonic) isValidInitialEntropy(strength entropyStrength) {
	    return (strength%32 > 0 || strength < 128 || strength > 256)
	}
```
Additionally you'll have an isolated method to run simple unit tests against it :)

Reply:
@adriacidre yes I think I'll try to do another PR for the refactoring. For now I just wanted to fix a "bug" and add tests where needed but I didn't want to change too much code.